### PR TITLE
[server/client] Implement secret mode ask

### PR DESCRIPTION
**Pull request by JungoKim**

### CODE REVIEW

File: client/src/app/components/my-card-list.jsx [hunk 1]
```diff
@@ -13,27 +14,102 @@ var AskResult = require('./ask-result.jsx');
...
+
+var AskHeader = React.createClass({
   render: function () {
-    console.log("!!!!!!Author rendered");
+     var styles = {
+      author: {
+        width:'70%',
+        float: 'left',
+      },
+      crownContainer: {
```
Comment (luckypapa) [inline client/src/app/components/my-card-list.jsx]: Perhaps .... unnecessary?

File: client/src/app/components/my-card-list.jsx [hunk 2]
```diff
@@ -13,27 +14,102 @@ var AskResult = require('./ask-result.jsx');
...
     console.log(this.props.userName.S);
     console.log(this.props.date.S);
+    var secret = this.props.secret === undefined ? "none" : this.props.secret.S;
     var userId = this.props.userId.S;
-    var author = this.props.userName.S;
+    var author = secret === "none" ? this.props.userName.S : secret;
     var yearMonthDay = new Date(parseInt(this.props.date.S)).toDateString();
     var time = new Date(parseInt(this.props.date.S)).toTimeString().split(' ')[0];
+    var readable = this.readableDate(this.props.date.S);
+    var rank = this.props.rank;
```
Comment (luckypapa) [inline client/src/app/components/my-card-list.jsx]: This is also unnecessary.

File: client/src/app/components/my-card-list.jsx [hunk 3]
```diff
@@ -181,7 +257,14 @@ var MyCardList = React.createClass({
             key={ask.index.S}
             initiallyExpanded={true}
             style={styles.card} >
-            <Author userName={ask.userName} userId={ask.userId} date={ask.date} />
+            <AskHeader
+              userName={ask.userName}
+              userId={ask.userId}
+              date={ask.date}
+              rank={ask.rank}
```
Comment (luckypapa) [inline client/src/app/components/my-card-list.jsx]: This is also unnecessary.

**Comment by JungoKim** [comment]
Please review again

File: client/src/app/components/ask-header.jsx [hunk 4]
```diff
@@ -27,12 +27,12 @@ var AskHeader = React.createClass({
...
     console.log(this.props.date.S);
+    var secret = this.props.secret === undefined ? "none" : this.props.secret.S;
     var userId = this.props.userId.S;
-    var author = this.props.userName.S;
+    var author = secret === "none" ? this.props.userName.S : secret;
     var yearMonthDay = new Date(parseInt(this.props.date.S)).toDateString();
     var time = new Date(parseInt(this.props.date.S)).toTimeString().split(' ')[0];
     var readable = this.readableDate(this.props.date.S);
-    var profile_photo = "http://graph.facebook.com/"+userId+"/picture?type=small";
     var rank = this.props.rank;
```
Comment (Myoungsun-Lee) [inline client/src/app/components/ask-header.jsx]: 불필요한 변수입니다.

**Comment by Myoungsun-Lee** [comment]
+1

**Pull request merged**